src/OrchardCore/Orchard.Localization.Core/StringLocalizer.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Microsoft.Extensions.Localization;
+using Orchard.Localization.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Orchard.Localization.Core
+{
+    public class StringLocalizer : IStringLocalizer
```

Comment:
Also create an `IPluralStringLocalizer: IStringLocalizer` that implements the indexers for the plural forms.
Create an `IPluralViewLocalizer: IViewLocalizer` that embed an instance of the default `ViewLocalizer` to forward all the calls to it, and also has custom plural indexers.